granulate_utils/linux/cgroups/base_cgroup.py

```diff
@@ -0,0 +1,69 @@
+#
+# Copyright (c) Granulate. All rights reserved.
+# Licensed under the AGPL3 License. See LICENSE.md in the project root for license information.
+#
+
+from enum import Enum
+from pathlib import Path
+from typing import List
+
+PID_CGROUPS = Path("/proc/self/cgroup")
+CGROUPFS = Path("/sys/fs/cgroup")
+
+
+class SUBSYSTEMS(Enum):
+    memory = "memory"
+    cpu = "cpu,cpuacct"
+
+
+class BaseCgroup:
+    predefined_cgroups = ["kubepods", "docker", "ecs"]
+
+    def __init__(self) -> None:
+        self._verify_preconditions()
+
+    def _verify_preconditions(self) -> None:
+        assert self.subsystem in SUBSYSTEMS.__members__, f"{self.subsystem!r} is not supported"
+
+        # "/proc/$PID/cgroup" lists a process's cgroup membership.  If legacy
+        # cgroup is in use in the system, this file may contain multiple lines, one for each hierarchy.
+        # The entry for cgroup v2 is always in the format "0::$PATH"::
+        if len(PID_CGROUPS.read_text().split("\n")) == 2:
+            raise Exception("cgroup V2 is unsupported")
+
+    @property
+    def subsystem(self) -> str:
+        raise NotImplementedError()
+
+    def _get_cgroup(self) -> str:
+        for line in PID_CGROUPS.read_text().split("\n"):
+            hierarchy = line.strip().split(":")
+            if SUBSYSTEMS[self.subsystem].value == hierarchy[1]:
+                return hierarchy[2]
+        raise Exception(f"{self.subsystem!r} is not in found")
+
+    def move_to_cgroup(self, custom_cgroup: str, pid: int = 0) -> None:
```

Comment:
```suggestion
    def move_to_cgroup(self, custom_cgroup: str, tid: int = 0) -> None:
```

logically, threads are part of cgroups, not processes

Reply:
Document that if `tid==0`, Linux will use the current thread